main.go

```diff
@@ -812,6 +812,35 @@ func (task *TaskRun) ExecuteCommand(index int) *CommandExecutionError {
 	if err != nil {
 		return WorkerShutdown(err)
 	}
+
+	// the duration until the expiry is precisely -1 *
+	// the amount of time since the expiry
+	duration := -time.Since(time.Time(task.Definition.Expires))
+	commandLogURL, err := Queue.GetArtifact_SignedURL(
```

Comment:
I did this in case a task had private logs. If they are public, it shouldn't expire, but if they are private, the expiry should match the expiry of the artifact (assuming the worker clock is correct). Until now all logs have been public, but I saw it as safer not to assume they always will be. In the case of the log being protected, if you have the scopes to view the live log, then you should have the scopes to read the log (as the livelog artifact name is the log name + ".live").